bookwyrm/models/book.py

```diff
@@ -492,10 +492,34 @@ def hyphenated_isbn13(self):
         """generate the hyphenated version of the ISBN-13"""
         return hyphenator.hyphenate(self.isbn_13)
 
-    @property
-    def average_rating(self, user):
-        """generate average rating of a book"""
-        reviews = Review.privacy_filter(user).filter(book__parent_work__editions=self)
+    @classmethod
+    def average_rating(cls, include_parents=False, user=None, privacy=None):
+        """
+        Generate the average rating of a book
+
+        Args:
+            include_parents: allows the caller to determine if ratings
+            from other Editions of the Work should be included
+            user: if we're given a value for user, use privacy_filter
+            privacy: if we're given a value for privacy, use it in filtering the reviews
+        Returns:
+            an aggregation of the average rating of the book across its reviews.
+        """
+        # get the base review queryset which we will refine
+        # based on the arguments provided
+        if include_parents:
+            base_reviews = Review.objects.filter(
+                Q(book__parent_work__editions=cls) & Q(deleted=False) & Q(rating_gt=0)
+            )
+        else:
+            base_reviews = Review.objects.filter(Q(deleted=False) & Q(rating_gt=0))
+        if user:
+            reviews = base_reviews.privacy_filter(user).exclude(rating=None)
+        elif privacy:
+            reviews = base_reviews.filter(Q(privacy=privacy)).exclude(rating=None)
+        else:
+            reviews = base_reviews.exclude(rating=None)
+
         return reviews.aggregate(Avg("rating"))["rating__avg"]
 
```

Comment:
> I think using a method that takes a `Review` queryset (which can already be filtered with things like `privacy_filter` or `privacy="public"`) and returns an average rating would be a good solution.

The filter calls here have overlap with the initial filter in the `Book` view [here](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/views/books/books.py#L68-L70)

At this stage of the refactor, I am okay with leaving the `Book` view filter as is and then allowing a more elegant solution to come out of our iterations on this.

@mouse-reeve what do you think?